clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -468,7 +468,7 @@ private static int parseAcks(String acksString) {
             }
 
             int partition = partition(record, serializedKey, serializedValue, cluster);
-            int serializedSize = Records.LOG_OVERHEAD + Record.recordSize(serializedKey, serializedValue);
+            int serializedSize = EosLogEntry.entrySizeUpperBound(serializedKey, serializedValue);
```

Comment:
It probably isn't a good idea to expose `EosLogEntry` here. How about adding a `size` method to`LogEntry` and implementing it appropriately in `OldLogEntry`, and `EosLogEntry`?